File: backend/model/response/tag.go
package response

import (
	"time"

	"github.com/google/uuid"
)

type TagResponse struct {
	ID        uuid.UUID `json:"id"`
	Name      string    `json:"name"`
	ColorHex  string    `json:"color_hex"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type TagPaginationMeta struct {
	TotalData int `json:"totalData"`
	Page      int `json:"page"`
	Limit     int `json:"limit"`
	TotalPage int `json:"totalPage"`
}

type TagPaginationResponse struct {
	Data []TagResponse     `json:"data"`
	Meta TagPaginationMeta `json:"meta"`
}

File: frontend/src/types/response/tag.ts
export interface TagResponse {
  id: number;
  name: string;
  color_hex: string;
  created_at: string;
  updated_at: string;
}

export interface TagPaginationMeta {
  totalData: number;
  page: number;
  limit: number;
  totalPage: number;
}

export interface TagPaginationResponse {
  data: TagResponse[];
  meta: TagPaginationMeta;
}
